redfish: Add redfish schemas only from implementation and oem

serge4k opened the pull request:
Implemented logic inside the "redfish-gen" tool to prepare schemas based on BMC implementation and Oem schemas necessary for the "Redfish Service Validator" tool.

Part of BBMC-1462

End-User-Impact: Low: (The BMC's redfish service may be validated
		by the "Redfish Service Validator" tool)

Commit:
redfish: Add redfish schemas only from implementation and oem

Implemented logic inside the "redfish-gen" tool to prepare schemas based on BMC implementation
and Oem schemas necessary for the "Redfish Service Validator" tool.

Part of BBMC-1462

End-User-Impact: Low: (The BMC's redfish service may be validated
		by the "Redfish Service Validator" tool)

Signed-off-by: Sergey Gulyaev <s.gulyaev@yadro.com>

ikmsk10 commented on tools/redfish-gen/redfish_gen/redfish_schema.py:
```diff
@@ -0,0 +1,543 @@
+## SPDX-License-Identifier: Apache-2.0
```
Looks like you have copied the redfish_schema.py from redfish_node.py, and now we have there are too many code-repeats.
Would it be better to import already existing redfish_node.py and inherit RedfishNode class with overriding required methods?

I have high doubt the way you picked is correct.

serge4k replied:
As discussed with @ikmsk10, I created the RedfishBase class with command functionality as super class for the RedfishNode and RedfishSchema.

ikmsk10 commented on tools/redfish-gen/redfish_gen/redfish_schema.py:
```diff
@@ -0,0 +1,543 @@
+## SPDX-License-Identifier: Apache-2.0
+## Copyright (C) 2022, KNS Group LLC (YADRO)
```
Please, specify correct copyright, today is 2023 year :)

ikmsk10 reviewed (changes requested):
Code Codacy has several warnings, please fix them.

Commit:
redfish-gen: Create RedfishBase class with common methods

Create common Redfish class with common functionality to
share same Redfish related functionaly in the 'redfish-gen'
tool.

End-User-Impact: None: (Code optimization without change
			functionality)

Signed-off-by: Sergey Gulyaev <s.gulyaev@yadro.com>

Commit:
redfish-gen: Move common methods to the RefishBase

Create common Redfish class with common functionality to
share same Redfish related functionaly in the 'redfish-gen'
tool.

Part of BBMC-1462

End-User-Impact: None: (Code optimization without change
			functionality)

Signed-off-by: Sergey Gulyaev <s.gulyaev@yadro.com>

Commit:
redfish-gen: Fixed Redfish-Service-Validator error in @odata.id

Create common Redfish class with common functionality to
share same Redfish related functionaly in the 'redfish-gen'
tool.

Part of BBMC-1462

End-User-Impact: None: (Code optimization without change
                        functionality)

Signed-off-by: Sergey Gulyaev <s.gulyaev@yadro.com>
